lib/mixins/flowdock.ts

```diff
@@ -0,0 +1,21 @@
+import * as request from 'request-promise';
+import { FlowdockInboxItem } from './flowdock-types';
+
+export class FlowdockAdapter {
+    public postToInbox(item: FlowdockInboxItem) {
```

Comment:
Why use this as a mixin, rather than a standalone object that VersionBot uses?

You could have `this.flowdock = new FlowdockAdapter()` in the VersionBot constructor, call `this.flowdock.postToInbox()` to get at it, get the same effect with less magic. Feels like a good case for composition over inheritance.